jujugui/static/gui/src/app/components/status/status.js

```diff
@@ -140,43 +163,115 @@ class Status extends React.Component {
+  */
+  _generateFilters() {
+    const options = ['none'].concat(this.STATUS_ORDER).map(status => {
+      return (
+        <option className="status-view__filter-option"
+          key={status}
+          value={status}>
+          {status}
+        </option>);
+    });
+    return (
+      <select className="status-view__filter-select"
+        onChange={this._handleFilterChange.bind(this)}>
+        {options}
+      </select>);
+  }
+
+  /**
     Generate the model fragment of the status.
     @param {Object} model The model attributes.
     @returns {Object} The resulting element.
   */
   _generateModel(model) {
+    const db = this.props.db;
     return (
-      <juju.components.BasicTable
-        headers={[{
-          content: 'Model',
-          columnSize: 3
-        }, {
-          content: 'Cloud/Region',
-          columnSize: 3
-        }, {
-          content: 'Version',
-          columnSize: 3
-        }, {
-          content: 'SLA',
-          columnSize: 3
-        }]}
-        key="model"
-        rows={[{
-          columns: [{
-            columnSize: 3,
-            content: model.environmentName
+      <div key="model">
+        <div className="twelve-col no-margin-bottom">
+          <div className="eight-col">
+            <h2>
+              {model.environmentName}
+            </h2>
+          </div>
+          <div className="status-view__filter-label two-col">
+            Filter status:
+          </div>
+          <div className="status-view__filter two-col last-col">
+            {this._generateFilters()}
+          </div>
+        </div>
+        <juju.components.BasicTable
+          headers={[{
+            content: 'Cloud/Region',
+            columnSize: 2
           }, {
-            columnSize: 3,
-            content: `${model.cloud}/${model.region}`
+            content: 'Version',
+            columnSize: 2
           }, {
-            columnSize: 3,
-            content: model.version
+            content: 'SLA',
+            columnSize: 1
           }, {
-            columnSize: 3,
-            content: model.sla
-          }],
-          key: 'model'
-        }]} />);
+            content: 'Applications',
+            columnSize: 2
+          }, {
+            content: 'Remote applications',
+            columnSize: 2
+          }, {
+            content: 'Units',
+            columnSize: 1
+          }, {
+            content: 'Machines',
+            columnSize: 1
+          }, {
+            content: 'Relations',
+            columnSize: 1
+          }]}
+          rows={[{
+            columns: [{
+              columnSize: 2,
+              content: `${model.cloud}/${model.region}`
+            }, {
+              columnSize: 2,
+              content: model.version
+            }, {
+              columnSize: 1,
+              content: model.sla
+            }, {
+              columnSize: 2,
+              content: (db.services ? db.services.size() : 0).toString()
```

Comment:
All these calls to size are not ok. The numbers that we want to display are the numbers of real entities, with uncommitted ones tripped out.This is already done when generating the other parts of the status pane in _generateStatus, so in theory this method here could take the numbers as calculated there. Lets use the db just there in a single place. generateModel could just take a dict like {numApplications: 2, numMachines: 3} etc.